chasm/lib/scheduler/scheduler.go

```diff
@@ -599,3 +635,64 @@ func (s *Scheduler) validateConflictToken(token []byte) bool {
 	current := s.generateConflictToken()
 	return bytes.Equal(current, token)
 }
+
+// SearchAttributes returns the Temporal-managed key values for visibility.
+func (s *Scheduler) SearchAttributes(chasm.Context) []chasm.SearchAttributeKeyValue {
+	return []chasm.SearchAttributeKeyValue{
+		chasm.SearchAttributeTemporalSchedulePaused.Value(s.Schedule.State.Paused),
+		chasm.SearchAttributeTemporalNamespaceDivision.Value(scheduler.NamespaceDivision),
+	}
+}
+
+// Memo returns the scheduler's info block for visibility.
+func (s *Scheduler) Memo(
+	ctx chasm.Context,
+) map[string]chasm.VisibilityValue {
+	newInfo, err := s.GetListInfo(ctx)
+	if err != nil {
+		// Unable to retrieve list info. Return nil to skip memo update.
+		// Error will be logged once loggers are available in CHASM.
+		return nil
+	}
+
+	newInfoPayload, err := newInfo.Marshal()
+	if err != nil {
+		// Unable to marshal list info. Return nil to skip memo update.
+		// Error will be logged once loggers are available in CHASM.
+		return nil
+	}
+
+	return map[string]chasm.VisibilityValue{
+		visibilityMemoFieldInfo: chasm.VisibilityValueByteSlice(newInfoPayload),
```

Comment:
This approach may cause extra visibility writes because proto marshalling is not deterministic and you may be returning a different byte slice every time you marshal.

I would suggest removing `VisibilityValueByteSlice` from the library so others don't fall into this trap and adding a `VisiblityValueProtoMessage` that will use `proto.Equal` for the comparison.